scala/scala-impl/src/org/jetbrains/plugins/scala/project/notification/HydraArtifactsNotificationProvider.scala

```diff
@@ -0,0 +1,49 @@
+package org.jetbrains.plugins.scala.project.notification
+
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.ui.{EditorNotificationPanel, EditorNotifications}
+import com.intellij.openapi.util.Key
+import org.jetbrains.plugins.scala.compiler.{CompileServerManager, HydraCompilerSettings}
+import org.jetbrains.plugins.scala.project.notification.HydraArtifactsNotificationProvider._
+import org.jetbrains.plugins.scala.settings.HydraApplicationSettings
+import org.jetbrains.plugins.scala.project.ProjectExt
+
+/**
+  * @author Maris Alexandru
+  */
+class HydraArtifactsNotificationProvider(project: Project, notifications: EditorNotifications)
+  extends AbstractNotificationProvider(project, notifications) {
+
+  override def getKey = ProviderKey
+
+  override protected def hasDeveloperKit(module: Module): Boolean = {
+    val downloadedScalaVersions = HydraApplicationSettings.getInstance().getDownloadedScalaVersions
+    val hydraSettings = HydraCompilerSettings.getInstance(project)
+    val scalaVersions = for {
+      module <- project.scalaModules
+      scalaVersion <- module.sdk.compilerVersion
+    } yield scalaVersion
+
+    if(hydraSettings.isHydraEnabled) {
+      scalaVersions.forall(downloadedScalaVersions.contains(_))
+    } else {
+      true
+    }
+  }
```

Comment:
The implementation here got me wonder: what happens if a project has two modules that use two different Scala versions (say 2.12.2 and 2.12.3)?

Reply:
First I get the scala versions from all modules and then I get the already downloaded versions.  Then I look if all the scala versions in the project exist in the downloaded versions.

`scalaVersions.forall(downloadedScalaVersions.contains(_))`

Reply:
Let's say that both Hydra for Scala 2.12.2 and 2.12.3 need to be downloaded, by clicking on the "Download" button will the artifacts for both versions be downloaded? If that's the case, it's awesome!

Reply:
yes...both version will be downloaded

Reply:
Me happy :)